packages/util/src/environment.ts

```diff
@@ -72,6 +72,24 @@ export function isBrowser(): boolean {
   return typeof self === 'object' && self.self === self;
 }
 
+/**
+ * Detect browser extensions (Chrome and Firefox at least).
+ */
+interface BrowserRuntime {
+  id?: unknown;
+}
+declare const chrome: { runtime?: BrowserRuntime };
+declare const browser: { runtime?: BrowserRuntime };
+export function isBrowserExtension(): boolean {
+  const runtime =
```

Comment:
It looks like we're using TypeScript >= 3.7. Can't we use optional chaining? Something like this would be far more readable:

```ts
return chrome?.runtime?.id !== undefined || browser?.runtime?.id !== undefined;
```

Reply:
I think TypeScript's optional chaining is less defensive than my code (it just does null/undefined checks instead of verifying it's of type object), and I'd like to be as defensive as possible to e.g. deal with the case that the developer just happens to have a `browser` variable but it's a `string` instead of `object` or whatever.